### PR TITLE
refactor metadata for multiple roles

### PR DESCRIPTION
**Pull request by whunter**
add admin option to route and all items to admin_csv_stuff

add admin to doi link and begin handling

admin and reg user dataset downloads

oops. needs to be == not =

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -144,7 +144,8 @@ def dataset_download
     time_stamp = DateTime.now.strftime('%Q')
     download_generator = DownloadGenerator.new(time_stamp)
     download_generator.make_archive
-    download_generator.generate_dataset_download(params[:id], request.base_url)
+    admin_download = (!params[:admin].blank? && params[:admin] == "admin")
```
Comment (tingtingjh) [inline app/controllers/collections_controller.rb]: Is "!params[:admin].blank?" redundant?

File: app/models/generic_file.rb [hunk 2]
```diff
@@ -10,6 +10,31 @@ def csv_headers
     ['id', 'mime_type', 'file_size', 'filename', 'checksum', 'depositor', 'title', 'resource_type', 'keyword', 'creator', 'rights', 'contributor', 'abstract', 'date_created', 'subject', 'language', 'identifier', 'location', 'related_url', 'provenance', 'in_collections']
   end
 
+  def admin_csv_headers
```
Comment (tingtingjh) [inline app/models/generic_file.rb]: Move to admin_csv_stuff Module?

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (whunter) [reply app/controllers/collections_controller.rb]: Yeah, you're probably right. I'll simplify that.

File: app/models/generic_file.rb (hunk 2 above)
Comment (whunter) [reply app/models/generic_file.rb]: The csv headers for Collections and GenericFiles are different. So if we moved them out of the models we'd have to have two different methods, wouldn't we?
If it's in the model we can just pass the class into the `write_csv` method in the DownloadGenerator class and call:
`item_class.new.admin_csv_headers` and not have to check which type it is.

Comment (tingtingjh) [reply app/models/generic_file.rb]: Then keep the code as they are.

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (tingtingjh) [reply app/controllers/collections_controller.rb]: Nevermind. I am merging now.

**Pull request merged**